daemon/state.go

```diff
@@ -144,21 +144,25 @@ func (d *Daemon) restoreOldEndpoints(dir string, clean bool) (*endpointRestoreSt
 
 		state.restored = append(state.restored, ep)
 
-		delete(existingEndpoints, ep.IPv4.String())
-		delete(existingEndpoints, ep.IPv6.String())
+		if existingEndpoints != nil {
+			delete(existingEndpoints, ep.IPv4.String())
+			delete(existingEndpoints, ep.IPv6.String())
+		}
 	}
 
 	log.WithFields(logrus.Fields{
 		"count.restored": len(state.restored),
 		"count.total":    len(possibleEPs),
 	}).Info("Endpoints restored")
 
-	for hostIP, info := range existingEndpoints {
-		if ip := net.ParseIP(hostIP); !info.IsHost() && ip != nil {
-			if err := lxcmap.DeleteEntry(ip); err != nil {
-				log.WithError(err).Warn("Unable to delete obsolete endpoint from BPF map")
-			} else {
-				log.Debugf("Removed outdated endpoint %d from endpoint map", info.LxcID)
+	if existingEndpoints != nil {
+		for hostIP, info := range existingEndpoints {
```

Comment:
This loop should no-op when the map is nil, but perhaps you wanted to make it clear that it did nothing when nil?